### PR TITLE
refactor: reuse Argo CD types to manage manifest generation

### PR DESCRIPTION
**Pull request by alexmt**
PR changes kargo-render to use Argo CD settings schema & rendering implementation.

* [x] Code changes
* [x] Tests
* [x] Docs changes

### CODE REVIEW

File: schema.json
Comment (krancour) [inline schema.json:64]: Do we have some other method available to validate the config management object?

Even if so... I know how yucky it would be to reverse engineer a JSON schema from Argo CD, but what I really wanted with this schema is that we would eventually publish it somewhere public and then leverage it for context help.

If we have a decent schema, it would be easy, for instance, to create a VS Code plugin that recognizes `kargo-render.yaml` and provides context help as you edit.

File: service_test.go
Comment (krancour) [inline service_test.go:21]: Nit: Would be good to assert that the replacement for these three (`renderFn`) gets a default value when `NewService()` is invoked.

**Comment by krancour** [comment]
Have a few Qs, but overall, looks great to me.

File: docs/docs/20-environment-branches.md
Comment (alexmt) [reply docs/docs/20-environment-branches.md]: fixed

File: docs/docs/10-overview.md
Comment (alexmt) [reply docs/docs/10-overview.md]: Agreed, restored go-module how to.

File: schema.json
Comment (alexmt) [reply schema.json]: I've realized we can re-use application CRD definition. Added `./hack/fetch-argocd-schema.sh` that generates schema from CRD.

File: service_test.go
Comment (alexmt) [reply service_test.go:21]: done

**Comment by alexmt** [comment]
Thank for review @krancour PTAL

File: schema.json
Comment (krancour) [reply schema.json]: Nice

**Comment by krancour** [comment]
Just making a note here of our offline discussion that there seems to be a schema issue with the configManagement section. It currently needs to validate oneOf, and this seems to be an issue when you accidentally validate more than oneOf. Should it be anyOf?

File: schema.json
Comment (krancour) [inline schema.json:65]:
```suggestion
			"anyOf": [{
```

I think this change is needed, but last I checked (yesterday), this wasn't passing, but I think my editor has the old schema cached. Will play with it some more...

Comment (alexmt) [reply schema.json:72]: Thanks for catching issue with schema. Issue is fixed now - added tests that cover required scenarios.

**Comment by alexmt** [comment]
Thanks for review @krancour  PTAL

File: schema.json
Comment (krancour) [inline schema.json:71]: I wasn't familiar with this and had to look it up. Still having trouble wrapping my head around it, but I see you added test cases for this, so I am confident this is right.

File: config_test.go
Comment (alexmt) [reply config_test.go:174]: If config management is not specified explicitly, then repo server/ argocd auto-detects it based on directory content. ( E.g. if directory contains `kustomization.yaml` it is Kustomize)

Comment (krancour) [reply config_test.go:174]: Ah right. Sorry! Totally forgot about that.

LGTM

**Review by krancour — APPROVED** [review]

**Comment by krancour** [comment]
Can we please merge this after #212 if possible?

I want to squeeze out one more release before altering config schema. @evgeny-goldin and @shelby-moore want #212 and I want to get it into their hands before also having to adapt to this change.

**Comment by alexmt** [comment]
@krancour sure. Just reviewed and approved https://github.com/akuity/kargo-render/pull/212

Waiting for it to be merged

**Comment by krancour** [comment]
> Waiting for it to be merged

Thank you @alexmt! Merged and rc.33 cut.

I'll let you do the honors on this one whenever you're ready.

**Pull request merged**
